### PR TITLE
Throw on invalid public key when building Ethereum Address

### LINKED ISSUE #216: Always use uncompressed public key in Ethereum::Address

### PR DESCRIPTION
**Pull request by alejandro-isaza**
Fix #216

### CODE REVIEW

File: src/Ethereum/Address.cpp [hunk 1]
```diff
@@ -37,6 +37,9 @@ Address::Address(const std::vector<uint8_t>& data) {
 }
 
 Address::Address(const PublicKey& publicKey) {
+    if (publicKey.type() != PublicKeyType::secp256k1Extended) {
+        throw std::invalid_argument("Ethereum::Address needs an extended SECP256k1 public key.");
```
Comment (vikmeup) [inline src/Ethereum/Address.cpp]: I also wonder if we do this for Ethereum, should we do for bitcoin as well?

Comment (alejandro-isaza) [reply src/Ethereum/Address.cpp]: done

**Comment by hewigovens** [comment]
User may expect that we uncompress public key inside `Ethereum::Address`, because we return compressed public in `HDWallet::getPublicKeyFromExtended`

**Comment by alejandro-isaza** [comment]
You can't uncompress a compressed key.

**Comment by hewigovens** [comment]
https://github.com/TrustWallet/wallet-core/blob/master/trezor-crypto/include/TrezorCrypto/ecdsa.h#L77

**Comment by alejandro-isaza** [comment]
Interesting...ok

**Pull request merged**